### PR TITLE
Add senior software developer job description

### PR DESCRIPTION
**Pull request by lwlsns**
This pull request introduces the job description for a Senior Software Developer. The document outlines the key responsibilities, and requirements for the role. It is intended to serve as a template and basis for creating other developer job descriptions in the future.

The intention is that the fluffy bit at the start can be tailored to suit each area but the person specification, and how we asses would remain largely unchanged (except for specific tech stacks). 

Feedback on the overall structure and content is appreciated.
Suggestions for additional sections or details to enhance the job description are welcome.

### CODE REVIEW

File: software-development-roles/software-developer/senior-developer.md [hunk 1]
```diff
@@ -0,0 +1,153 @@
+
+
+# Senior Software Developer
+**Location:** National\*  
+**Closing Date:** [Insert Date]  
+**Interviews:** Expected w/c [Insert Week]
+
+**Grade:** G6
```
Comment (carlosMoJ) [inline software-development-roles/software-developer/senior-developer.md:8]: sorry if I'm out of the loop/this is changing, but aren't seniors grade 7?

File: software-development-roles/software-developer/senior-developer.md [hunk 2]
```diff
@@ -0,0 +1,153 @@
...
+
+- **Technologies:**
+  We use a diverse range of technologies, and we’re seeking individuals who specialize in one or more and are eager to learn new languages and frameworks. Our tech stack includes:
+  
+  - Application Code: Ruby, Kotlin, Java, Python, NodeJS, PHP, Wordpress
+  - CI/CD: Circle CI, GitHub Actions
+  - Monitoring: Sentry
+  - Containerization: Kubernetes, Docker
+  - Cloud Infrastructure: AWS
+  - Web Interfaces: GOV.UK Design System
```
Comment (carlosMoJ) [inline software-development-roles/software-developer/senior-developer.md:58]: is it worth adding a mention to MoJ Design system?

File: software-development-roles/software-developer/senior-developer.md (hunk 1 above)
Comment (lwlsns) [reply software-development-roles/software-developer/senior-developer.md]: Thanks @carlosMoJ, I've changed the grade to G7

File: software-development-roles/software-developer/senior-developer.md (hunk 2 above)
Comment (lwlsns) [reply software-development-roles/software-developer/senior-developer.md]: Added 👍

**Review by carlosMoJ — APPROVED** [review]
Looks good to me!

**Pull request merged**